source/game.stationmap.bmx

```diff
@@ -2428,128 +3163,94 @@ Type TStationMap Extends TOwnedGameObject {_exposeToLua="selected"}
-		reachBefore = self.reach
+		'sum of this can be > a maps population (eg antenna in bavaria + cable network in bavaria > bavaria's population)
+		self._reachedAntennaPopulation = GetStationMapCollection().GetAntennaPopulation(owner)
+		self._reachedCableNetworkUplinkPopulation = GetStationMapCollection().GetCableNetworkUplinkPopulation(stations)
+		self._reachedSatelliteUplinkPopulation = GetStationMapCollection().GetSatelliteUplinkPopulation(stations)
 
-		If cheatedMaxReach
-			reach = GetStationMapCollection().population
+		'so coverage of all potential station types needs to be calculated individually
+		'
+		'as soon as:
+		'- a satellite is used, whole map's population is reached
+		'- a cable network is used, antenna population of the section can be ignored
+		'- only antennas are used, their covered population is the total population
+		self._reachedPopulation = 0
+		if self._reachedSatelliteUplinkPopulation > 0
+			'do NOT use "= self._reachedSatelliteUplinkPopulation" as 
+			'this value depends on the satellite share on the sections
+			self._reachedPopulation = GetStationMapCollection().GetPopulation()
+		ElseIf self._reachedCableNetworkUplinkPopulation > 0
+			For Local section:TStationMapSection = EachIn GetStationMapCollection().sections
+				If section.GetCableNetworkUplinkPopulation() > 0
+					self._reachedPopulation :+ section.GetPopulation()
+				Else
+					self._reachedPopulation :+ section.GetAntennaPopulation(owner)
+				EndIf
+			Next
 		Else
-			If TStationMapCollection.populationReceiverMode = TStationMapCollection.RECEIVERMODE_SHARED
-				Throw "RecalculateAudienceSum: Todo"
-			ElseIf TStationMapCollection.populationReceiverMode = TStationMapCollection.RECEIVERMODE_EXCLUSIVE
-				reach =  GetStationMapCollection().GetAntennaAudienceSum(owner)
-				reach :+ GetStationMapCollection().GetCableNetworkUplinkAudienceSum(stations)
-				reach :+ GetStationMapCollection().GetSatelliteUplinkAudienceSum(stations)
-				'print "RON: antenna["+owner+"]: " + GetStationMapCollection().GetAntennaAudienceSum(owner) + "   cable["+owner+"]: " + GetStationMapCollection().GetCableNetworkUplinkAudienceSum(stations) +"   satellite["+owner+"]: " + GetStationMapCollection().GetSatelliteUplinkAudienceSum(stations) + "   recalculated: " + reach
-			EndIf
+			self._reachedPopulation = self._reachedAntennaPopulation
 		EndIf
+		
 
-		reachMax = Max(reach, reachMax)
-		'current reach is updated now
-		reachInvalid = False
+		self._reachedAntennaReceivers = GetStationMapCollection().GetAntennaReceivers(owner)
+		self._reachedCableNetworkUplinkReceivers = GetStationMapCollection().GetCableNetworkUplinkReceivers(stations)
+		self._reachedSatelliteUplinkReceivers = GetStationMapCollection().GetSatelliteUplinkReceivers(stations)
+		
+
+		'update record
+		self.reachedPopulationMax = Max(self.reachedPopulationMax, self._GetReachedPopulation())
+		self.reachedReceiversMax = Max(self.reachedReceiversMax, self._GetReachedReceivers())
```

Comment:
Hier frage ich mich, ob da was schief gehen kann. In den Jahren, in denen der Antennenanteil stark sinkt...

Reply:
Der Code wird an sich nirgends genutzt (bisher). Kann also gern auch komplett raus.